internal/bundler/bundler.go

```diff
@@ -225,46 +224,6 @@ func parseFile(args parseArgs) {
 		args.options.JSX.Parse = true
 
 		source.Contents = translationsHack(args.options.SpinxAssetBaseURL, source.PrettyPath, source.Contents, &args.caches.TranslationsCache)
-		if args.options.SpinxHotReact && args.options.Platform == config.PlatformBrowser {
-			segments := strings.Split(source.PrettyPath, "/")
-			componentName := strings.Replace(segments[len(segments)-1], ".tsx", "", 1)
-			matched, _ := regexp.MatchString(`^[a-zA-Z0-9]+$`, componentName)
-			if matched {
-				hash := "1"
-				source.Contents = fmt.Sprintf(`
```

Comment:
Changing the source string was bad for sourcemap accuracy. esbuild (rightly) included HMR boilerplate in the maps, but there are no equivalent lines in the original source, so everything ended up "off" by a few lines.